python/pyarrow/_dataset.pyx

```diff
@@ -73,10 +73,58 @@ cdef class FileFormat:
         return self.wrapped
 
 
+cdef class ParquetFileFormatReaderOptions:
+    cdef:
+        CParquetFileFormatReaderOptions* options
+
+    def __init__(self, ParquetFileFormat fmt):
```

Comment:
Since the ParquetFileFormatOptions are a member of the ParquetFileFormat, I suppose it is unavoidable to need to pass the format here on construction? 
But it gives a bit a strange pattern for the user to construct those options

Reply:
Although I see you can pass those as a dict to `ParquetFileFormat`, which alleviates that concern a bit. 

The pattern that is eg used in the csv reader would more look like this:

```
ParquetFileFormat(reader_options=ParquetFileFormatReaderOptions(dict_columns=[...]))
```

Reply:
I can use this pattern but I think I'll wait for a follow up